source/common/http/conn_manager_impl.h

```diff
@@ -334,44 +338,42 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
 
     // All state for the stream. Put here for readability.
     struct State {
-      State()
-          : codec_saw_local_complete_(false), codec_encode_complete_(false),
-            on_reset_stream_called_(false), is_zombie_stream_(false), successful_upgrade_(false),
-            is_internally_destroyed_(false), is_internally_created_(false), is_tunneling_(false),
-            decorated_propagate_(true), deferred_to_next_io_iteration_(false),
-            deferred_end_stream_(false) {}
-
       // It's possibly for the codec to see the completed response but not fully
       // encode it.
-      bool codec_saw_local_complete_ : 1; // This indicates that local is complete as the completed
-                                          // response has made its way to the codec.
-      bool codec_encode_complete_ : 1;    // This indicates that the codec has
-                                          // completed encoding the response.
-      bool on_reset_stream_called_ : 1;   // Whether the stream has been reset.
-      bool is_zombie_stream_ : 1;         // Whether stream is waiting for signal
-                                          // the underlying codec to be destroyed.
-      bool successful_upgrade_ : 1;
+      bool codec_saw_local_complete_{}; // This indicates that local is complete as the completed
+
+      // response has made its way to the codec.
+      bool codec_encode_complete_{};  // This indicates that the codec has
+                                      // completed encoding the response.
+      bool on_reset_stream_called_{}; // Whether the stream has been reset.
+      bool is_zombie_stream_{};       // Whether stream is waiting for signal
+                                      // the underlying codec to be destroyed.
+
+      // Whether the upgrade request has be accepted or rejected by the filter chain.
+      // Both are false if the request is not an upgrade request.
+      bool upgrade_accepted_{};
+      bool upgrade_rejected_{};
```

Comment:
This is most because the implementation details' difference. In the previous implmentation, the upgraded accepted result will be reported by the `upgradeFilterChainCreated` callback. The upgrade rejected result will be reported by the return value of the `createFilterChain`.

Now, we report all upgrade results with same callback. So, two statuses are necessary. (or an optional bool should be used, but I think two bool flag would much cheaper and easy to use, they are private statuses anyway.)

Reply:
can't we now remove this?  it doesn't appear to be used.

Reply:
right, I forgot it